PWGJE/Tasks/nucleiWithFastJet.cxx

```diff
@@ -0,0 +1,1500 @@
+          }
+        }
+
+        if (track.mcParticle().pdgCode() == -PDGTriton) {
+          jetHist.fill(HIST("recmatched/antiTriton/pt/PtantiTriton"), track.mcParticle().pt());
+          if (jetFlag) {
+            jetHist.fill(HIST("recmatched/antiTriton/pt/PtantiTriton_jet"), track.mcParticle().pt());
+          } else {
+            jetHist.fill(HIST("recmatched/antiTriton/pt/PtantiTriton_outJet"), track.mcParticle().pt());
+          }
+        }
+
+      } // DetJet
+    }   // jet constituents
+
+  } // process
+
+  int nprocessSimJEEvents = 0;
+  void processGenMatched(aod::JMcCollision const& collision,
+                         /*soa::SmallGroups<soa::Join<aod::JMcCollisionLbs, aod::JCollisions>> const& recocolls,*/
+                         JetMCDetTable const&, JetMCPartTable const& mcpjets, aod::JMcParticles const& mcParticles)
+  {
+
+    if (cDebugLevel > 0) {
+      nprocessSimJEEvents++;
+      if ((nprocessSimJEEvents + 1) % 100000 == 0)
+        LOG(debug) << "Jet Events: " << nprocessSimJEEvents;
+    }
+    if (fabs(collision.posZ()) > 10)
+      return;
+
+    jetHist.fill(HIST("genmatched/vertexZ"), collision.posZ());
+
+    bool INELgt0 = false;
+    for (const auto& mcParticle : mcParticles) {
+      if (fabs(mcParticle.eta()) < cfgtrkMaxEta) {
+        INELgt0 = true;
+        break;
+      }
+    }
+    if (!INELgt0)
+      return;
+
+    std::vector<double> mcdJetPt{};
+    std::vector<double> mcdJetPhi{};
+    std::vector<double> mcdJetEta{};
+    std::vector<double> mcpJetPt{};
+    std::vector<double> mcpJetPhi{};
+    std::vector<double> mcpJetEta{};
+
+    for (auto& mcpjet : mcpjets) {
+      jetHist.fill(HIST("genmatched/hGenJetPt"), mcpjet.pt());
+      if (!mcpjet.has_matchedJetGeo())
+        continue;
+      jetHist.fill(HIST("genmatched/hGenJetPtMatched"), mcpjet.pt());
+      for (auto& mcdjet : mcpjet.template matchedJetGeo_as<JetMCDetTable>()) {
+        if (!mcdjet.has_matchedJetGeo())
+          continue;
+        mcdJetPt.push_back(mcdjet.pt());
+        mcdJetPhi.push_back(mcdjet.phi());
+        mcdJetEta.push_back(mcdjet.eta());
+        mcpJetPt.push_back(mcpjet.pt());
+        mcpJetPhi.push_back(mcpjet.phi());
+        mcpJetEta.push_back(mcpjet.eta());
+
+        jetHist.fill(HIST("genmatched/hRecJetPt"), mcpjet.pt());
+        jetHist.fill(HIST("genmatched/hRecJetWithGenPt"), mcdjet.pt());
+        jetHist.fill(HIST("genmatched/hRecMatchedJetPt"), mcpjet.pt(), mcpjet.pt() - mcdjet.pt());
+        jetHist.fill(HIST("genmatched/hRecMatchedJetPhi"), mcpjet.phi(), mcpjet.phi() - mcdjet.phi());
+        jetHist.fill(HIST("genmatched/hRecMatchedJetEta"), mcpjet.eta(), mcpjet.eta() - mcdjet.eta());
+
+      } // mcdJet
+    }   // mcpJet
+
+    for (const auto& mcParticle : mcParticles) {
+      if (fabs(mcParticle.eta()) > cfgtrkMaxEta)
+        continue;
+      // add pid later
+
+      bool jetFlag = false;
+      for (int iDJet = 0; iDJet < mcpJetPt.size(); iDJet++) {
```

Comment:
why dont you just loop over the mcp jets here and her eta and phi directly from the object?

Reply:
same with the process function above

Reply:
Thank you very much @nzardosh for the comments. Here, I loop over only the matched jets instead of looping over the all mcp jets.